src/test/java/com/revature/controllers/ReviewControllerTest.java

```diff
@@ -212,15 +237,50 @@ void testAddReview_Success() throws Exception {
 	}
 
 	@Test
-	@Disabled("Not yet implemented")
-	void testAddReview_Failure_NotLoggedIn() {
-		fail("Not yet implemented");
+	void testAddReview_Failure_NotLoggedIn() throws Exception {
+		int authorId = this.dummyUser.getId();
+		ReviewRequest newReview = new ReviewRequest(authorId, this.dummyProduct.getId(), this.dummyReview.getStars(),
+				this.dummyReview.getTitle(), this.dummyReview.getReviewMessage());
+
+		// Given the security changes, is simply not having a token equivalent to not being logged in?
+		String jsonContent = this.jsonReviewRequest.write(newReview).getJson();
+		MockHttpServletRequestBuilder request = post(this.MAPPING_ROOT).contentType(MediaType.APPLICATION_JSON)
+				.content(jsonContent).sessionAttr("user", this.dummyUser);
+		MockHttpServletResponse response = this.mvc.perform(request).andReturn().getResponse();
+
+		assertEquals(HttpStatus.FORBIDDEN.value(), response.getStatus());
 	}
 
 	@Test
-	@Disabled("Not yet implemented")
-	void testAddReview_Failure_ProductNotFound() {
-		fail("Not yet implemented");
+	void testAddReview_Failure_ProductNotFound() throws Exception {
+		int authorId = this.dummyUser.getId();
+		ReviewRequest newReview = new ReviewRequest(authorId, this.dummyProduct.getId(), this.dummyReview.getStars(),
+				this.dummyReview.getTitle(), this.dummyReview.getReviewMessage());
+		given(this.rServ.add(newReview)).willThrow(new ProductNotFoundException());
+
+		String jsonContent = this.jsonReviewRequest.write(newReview).getJson();
+		MockHttpServletRequestBuilder request = post(this.MAPPING_ROOT).contentType(MediaType.APPLICATION_JSON)
+				.content(jsonContent).sessionAttr("user", this.dummyUser)
+				.header(HttpHeaders.AUTHORIZATION, TOKEN);
+		MockHttpServletResponse response = this.mvc.perform(request).andReturn().getResponse();
+
+		assertEquals(HttpStatus.NOT_FOUND.value(), response.getStatus());
+	}
+	
+	@Test
+	void testAddReview_Failure_AlreadyReviewed() throws Exception {
+		int authorId = this.dummyUser.getId();
+		ReviewRequest newReview = new ReviewRequest(authorId, this.dummyProduct.getId(), this.dummyReview.getStars(),
+				this.dummyReview.getTitle(), this.dummyReview.getReviewMessage());
+		given(this.rServ.add(newReview)).willThrow(new DuplicateReviewException());
+
+		String jsonContent = this.jsonReviewRequest.write(newReview).getJson();
+		MockHttpServletRequestBuilder request = post(this.MAPPING_ROOT).contentType(MediaType.APPLICATION_JSON)
+				.content(jsonContent).sessionAttr("user", this.dummyUser)
+				.header(HttpHeaders.AUTHORIZATION, TOKEN);
+		MockHttpServletResponse response = this.mvc.perform(request).andReturn().getResponse();
+
+		assertEquals(HttpStatus.CONFLICT.value(), response.getStatus());
 	}
 
```

Comment:
It would be nice if we test for submitting a new review to fail if the provided user ID is not found, but I'm not sure if it will help with the test coverage that much